libsolutil/CMakeLists.txt

```diff
@@ -47,7 +47,7 @@ set(sources
 )
 
 add_library(solutil ${sources})
-target_link_libraries(solutil PUBLIC Boost::boost Boost::filesystem Boost::system range-v3 fmt::fmt-header-only nlohmann_json::nlohmann_json)
+target_link_libraries(solutil PUBLIC Boost::boost Boost::filesystem ${Boost_SYSTEM_LIBRARY} range-v3 fmt::fmt-header-only nlohmann_json::nlohmann_json)
```

Comment:
I am a bit uneasy about the usage of internal/undocumented Boost_SYSTEM_LIBRARY here. Couldn't we achieve the same with a generator expression, e.g., `$<TARGET_NAME_IF_EXISTS:Boost::system>`?

Reply:
The FindBoost module does document `Boost_<COMPONENT>_LIBRARY`[^1] while BoostConfig.cmake provides compatibility with the module[^2].

Only thing is if you don't want to rely on `Boost_SYSTEM_LIBRARY` remaining undefined. Can try a generator expression if preferred.

[^1]: https://cmake.org/cmake/help/latest/module/FindBoost.html#result-variables
[^2]: https://github.com/boostorg/boost_install/blob/boost-1.89.0/BoostConfig.cmake#L152-L157